endpoints/setuid.go

```diff
@@ -148,6 +163,62 @@ func NewSetUIDEndpoint(cfg *config.Configuration, syncersByBidder map[string]use
 	})
 }
 
+// extractGDPRInfo looks for the GDPR consent string and GDPR signal in the GPP query params
+// first and the 'gdpr' and 'gdpr_consent' query params second.
+func extractGDPRInfo(query url.Values) (gdpr.RequestInfo, error) {
+	var gdprSignal gdpr.Signal = gdpr.SignalNo
+	var gdprConsent string = query.Get("gdpr_consent")
+	var signalSet bool = false
+
+	// Signal
+	gppSID, err := stringutil.StrToInt8Slice(query.Get("gpp_sid"))
+	if err != nil {
+		return gdpr.RequestInfo{}, fmt.Errorf("Error parsing gpp_sid %s", err.Error())
+	}
+	if len(gppSID) > 0 && gppPrivacy.IsSIDInList(gppSID, gppConstants.SectionTCFEU2) {
+		gdprSignal = gdpr.SignalYes
+		signalSet = true
+	}
```

Comment:
if the length of `gppSID` is greater than 0 but `gppConstants.SectionTCFEU2` is not present in `gppSID`, should we be treating that as a `SignalNo` and not bother to consult the legacy query params? It looks like that is what we're doing in the cookie sync endpoint.

Reply:
Good catch. Logic was corrected

Reply:
Good catch. Logic was corrected